src/restconfig/src/main/java/org/geoserver/rest/resources/ResourceResource.java

```diff
@@ -229,10 +230,27 @@ private String href(String path, boolean isDir) {
         return href;
     }
 
-    public String formatHtmlLink(String link) {
+    private static String formatHtmlLink(String link) {
         return link.replaceAll("&", "&amp;");
     }
 
+    private static String formatPathURL(String input) {
```

Comment:
Ben, URLEncoder.encode also changes spaces into + instead of %20.
Is that something we want?